include/revng/Pipes/CompileModule.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Pipeline/ContainerSet.h"
+#include "revng/Pipeline/Context.h"
+#include "revng/Pipeline/Contract.h"
+#include "revng/Pipeline/GenericLLVMPipe.h"
+#include "revng/Pipeline/LLVMContainer.h"
+#include "revng/Pipeline/LLVMGlobalKindBase.h"
+#include "revng/Pipeline/Target.h"
+#include "revng/Pipes/IsolatedKind.h"
+#include "revng/Pipes/RevngPipes.h"
+#include "revng/Pipes/RootKind.h"
+#include "revng/Pipes/TemporaryFile.h"
+
+namespace pipeline {
+
+class CompileModulePipe {
+public:
+  static constexpr auto Name = "Compile Module Pipe";
+  std::array<ContractGroup, 1> getContract() const {
+    return { ContractGroup(Root,
+                           Exactness::DerivedFrom,
+                           0,
+                           Object,
+                           1,
+                           InputPreservation::Preserve) };
+  }
+  void
+  run(const Context &, LLVMContainer &TargetsList, TemporaryFile &TargetBinary);
```

Comment:
* `TargetsList` seems a bad name
* Would it make sense for `TargetsList` to be `const`?

Reply:
the compilation is operated as a llvm passmanager and that requires a non cost reference.